### ISSUE TITLE
Strange generation of nullable and non-nullable lists

### ISSUE THREAD

**Issue by meg4cyberc4t** [heftapp/graphql_codegen#309]
Hi.
I have this input in my schema.graphql.

```graphql
input TemplateDisciplineTopicContentInput {
  howStudyIt: String!
  infoContentBlocks: [InfoTemplateDisciplineTopicContentBlockInput!]!
  taskContentBlocks: [TaskTemplateDisciplineTopicContentBlockInput!]!
  testContentBlocks: [TestTemplateDisciplineTopicContentBlockInput!]! = []
  whyStudyIt: String!
}
```
You can see that "testContentBlocks" is not an updatable list with a default value. And "taskContentBlocks" is the same, but does not have a default value.
But they are generated in different ways, it confuses me exactly:
```dart
 List<Input$InfoDisciplineTopicContentBlockInput> get infoContentBlocks =>
      (_$data['infoContentBlocks']
          as List<Input$InfoDisciplineTopicContentBlockInput>);
  List<Input$TaskDisciplineTopicContentBlockInput> get taskContentBlocks =>
      (_$data['taskContentBlocks']
          as List<Input$TaskDisciplineTopicContentBlockInput>);
  List<Input$TestDisciplineTopicContentBlockInput>? get testContentBlocks =>
      (_$data['testContentBlocks']
```
"testContentBlocks" has become nullable.. but however, in the rest of the code, it is still considered non-null.
```dart
  TRes testContentBlocks(
          Iterable<Input$TestDisciplineTopicContentBlockInput> Function(
                  Iterable<
                      CopyWith$Input$TestDisciplineTopicContentBlockInput<
                          Input$TestDisciplineTopicContentBlockInput>>)
              _fn) =>
      call(
          testContentBlocks: _fn(_instance.testContentBlocks
              .map((e) => CopyWith$Input$TestDisciplineTopicContentBlockInput(
                    e,
                    (i) => i,
                  ))).toList());
```
-- This is an example where my linter indicates the use of a list without an operator ?.

As a result, an error occurs when building the application.

**Comment by meg4cyberc4t** [comment]
To be honest, I don't fully understand how the values in the default input are processed on the client side.
In the generated files I also see:
```dart
 factory Input$DisciplineTopicContentInput({
    required String howStudyIt,
    required List<Input$InfoDisciplineTopicContentBlockInput> infoContentBlocks,
    required List<Input$TaskDisciplineTopicContentBlockInput> taskContentBlocks,
    List<Input$TestDisciplineTopicContentBlockInput>? testContentBlocks,
    required String whyStudyIt,
  }) =>
      Input$DisciplineTopicContentInput._({
        r'howStudyIt': howStudyIt,
        r'infoContentBlocks': infoContentBlocks,
        r'taskContentBlocks': taskContentBlocks,
        if (testContentBlocks != null) r'testContentBlocks': testContentBlocks,
        r'whyStudyIt': whyStudyIt,
      });
```
But because of this nullable value, the generation of "_CopyWithImpl$Input$DisciplineTopicContentInput<TRes>" crashes

**Comment by meg4cyberc4t** [comment]
For enough context:
  
  graphql: ^5.2.0-beta.1
  graphql_flutter: ^5.1.2
  graphql_codegen: ^0.13.5

**Comment by budde377** [comment]
I've recently changed how input with default values are generated and it would seem like I've missed a few places. Let me investigate!

**Comment by meg4cyberc4t** [comment]
Hi. Is there any updates for me?)

**Comment by budde377** [comment]
Sorry Igor, I have been busy moving. I am to look at this this weekend!

**Comment by budde377** [comment]
A fix should be released now as `0.13.6`.

**Comment by meg4cyberc4t** [comment]
It looks great! Thanks!
